internal/zinc-core/src/main/scala/sbt/internal/inc/Stamp.scala

```diff
@@ -58,10 +59,9 @@ trait WithPattern { protected def Pattern: Regex }
 import java.lang.{ Long => BoxedLong }
 
 /** Define the hash of the file contents. It's a typical stamp for compilation sources. */
-final class Hash(val value: Array[Byte]) extends StampBase {
-  val hexHash = IOHash.toHex(value)
+final class Hash(val hexHash: String) extends StampBase {
```

Comment:
.. at the cost of type-safety (not all strings are hex strings; note the use of the name`hexString` rather than plain `value`). Previously this would fail-fast on bad data (e.g `"hash(gobbledygook)"`).

Also given each byte is transformed into 2 chars I wonder about the equality performance gain claim. Maybe String.equals is special-cased.